core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -475,6 +533,194 @@ object ConsumerGroupCommand extends Logging {
       new KafkaConsumer(properties)
     }
 
+    def resetOffsets(): Map[PartitionAssignmentState, Long] = {
+      val groupId = opts.options.valueOf(opts.groupOpt)
+      val (state, assignments) = collectGroupAssignment(groupId)
+      assignments match {
+        case None =>
+          // applies to both old and new consumer
+          printError(s"The consumer group '$groupId' does not exist.")
+          Map.empty
+        case Some(assignments) =>
+          state match {
```

Comment:
This match feels like overkill. Seems the only case we care about is if the group is empty. Otherwise, maybe we can print a generic error message. For example: "Assignments can only be reset if the group is empty, but the current state is $state".